bundles/org.openhab.binding.awattar/src/main/java/org/openhab/binding/awattar/internal/handler/AwattarBridgeHandler.java

```diff
@@ -128,54 +107,24 @@ void refreshIfNeeded() {
         }
     }
 
+    /**
+     * Refresh the data from the API.
+     */
     private void refresh() {
         try {
-            // we start one day in the past to cover ranges that already started yesterday
-            ZonedDateTime zdt = LocalDate.now(zone).atStartOfDay(zone).minusDays(1);
-            long start = zdt.toInstant().toEpochMilli();
-            // Starting from midnight yesterday we add three days so that the range covers the whole next day.
-            zdt = zdt.plusDays(3);
-            long end = zdt.toInstant().toEpochMilli();
-
-            StringBuilder request = new StringBuilder(url);
-            request.append("?start=").append(start).append("&end=").append(end);
-
-            logger.trace("aWATTar API request: = '{}'", request);
-            ContentResponse contentResponse = httpClient.newRequest(request.toString()).method(GET)
-                    .timeout(10, TimeUnit.SECONDS).send();
-            int httpStatus = contentResponse.getStatus();
-            String content = contentResponse.getContentAsString();
-            logger.trace("aWATTar API response: status = {}, content = '{}'", httpStatus, content);
-
-            if (httpStatus == OK_200) {
-                Gson gson = new Gson();
-                SortedSet<AwattarPrice> result = new TreeSet<>(Comparator.comparing(AwattarPrice::timerange));
-                AwattarApiData apiData = gson.fromJson(content, AwattarApiData.class);
-                if (apiData != null) {
-                    for (Datum d : apiData.data) {
-                        double netPrice = d.marketprice / 10.0;
-                        TimeRange timerange = new TimeRange(d.startTimestamp, d.endTimestamp);
-                        result.add(new AwattarPrice(netPrice, netPrice * vatFactor, netPrice + basePrice,
-                                (netPrice + basePrice) * vatFactor, timerange));
-                    }
-                    prices = result;
-                    updateStatus(ThingStatus.ONLINE);
-                } else {
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                            "@text/error.invalid.data");
-                }
-            } else {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                        "@text/warn.awattar.statuscode");
-            }
+            prices = awattarApi.getData();
+            updateStatus(ThingStatus.ONLINE);
+        } catch (ExecutionException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "@text/error.execution");
         } catch (JsonSyntaxException e) {
```

Comment:
If you move everything to the api class, also move error handing there. You can rename `EmptyResponseException` to `AwattarApiException` and catch that here.

Reply:
Is it possible to use `updateStatus` from there? It is a function of the Bridge superclass.

Reply:
What he was referring to was wrapping the `JsonSyntaxException` (which is an implementation detail of the API) into an API specific wrapper exception and catching that one here instead of `JsonSyntaxException`.

Reply:
No, you can't. My suggestion was something like

```
   } catch (AwattarApiException e) {
       updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
   }
```

and in the api class

```
   } catch (JsonSyntaxException e) {
       throw new AwattarApiException("@text/error.json");
   }
```

ans similar for the other exceptions, so that everything (the `InterruptedException` and others) except `AwattarApiException` is caught in the api implementation class.

Reply:
Ah ok, I actually has it like this but I wasn't sure if this work with exceptions too.

Sure will change it.